clang/tools/clang-offload-bundler/ClangOffloadBundler.cpp

```diff
@@ -589,36 +585,8 @@ class ObjectFileHandler final : public FileHandler {
       if (!BufOrErr)
         return createFileError(InputFileNames[I], BufOrErr.getError());
 
-      std::unique_ptr<MemoryBuffer> Buf = std::move(*BufOrErr);
```

Comment:
How non-file-scope asm blocks are handled?
I think we should also check that something like this is not allowed, otherwise ignoring doesn't seem to be a good option.
```c++
SYCL_EXTERNAL __asm__("foo");
```

Reply:
I didn't touch logic for non-file-scope asm. I just disabled only file-scope asm generation. There is a same behavior for cuda and openmp please take a look at the codegen code.

Reply:
in clang-offload-bundler I just removed workaround for this problem (propagation of file-scope asm to device code)